Formula/libarchive.rb

```diff
@@ -42,6 +42,9 @@ def install
 
     system "make", "install"
 
+    # fixes https://github.com/libarchive/libarchive/issues/1819
+    inreplace lib/"pkgconfig/libarchive.pc", "Libs.private: ", "Libs.private: -liconv "
+    inreplace lib/"pkgconfig/libarchive.pc", "Requires.private: iconv", ""
```

Comment:
```suggestion
    inreplace lib/"pkgconfig/libarchive.pc" do |s|
      s.gsub! "Libs.private: ", "Libs.private: -liconv "
      s.gsub! "Requires.private: iconv", ""
    end
```
`-liconv` might be macOS-only, though.